account-kit/smart-contracts/src/light-account/accounts/viem-base.ts

```diff
@@ -0,0 +1,344 @@
+      },
+      message: {
+        message: hashedMessage,
+      },
+      primaryType: "LightAccountMessage",
+    };
+  };
+
+  const prepareSignature = async (
+    message: string | TypedDataDefinition,
+    isTypedData: boolean = false,
+  ): Promise<{ shouldWrap: boolean; data: string | TypedDataDefinition }> => {
+    const messageHash = isTypedData
+      ? hashTypedData(message as TypedDataDefinition)
+      : hashMessage(message as string);
+
+    switch (version as string) {
+      case "v1.0.1":
+        return { shouldWrap: false, data: message };
+      case "v1.0.2":
+        throw new Error(
+          `Version ${String(version)} of LightAccount doesn't support 1271`,
+        );
+      case "v1.1.0":
+        return { shouldWrap: true, data: get1271Wrapper(messageHash, "1") };
+      case "v2.0.0":
+        return { shouldWrap: true, data: get1271Wrapper(messageHash, "2") };
+      default:
+        throw new Error(`Unknown version ${String(version)} of LightAccount`);
+    }
+  };
+
+  const formatSignature = (signature: Hex): Hex => {
+    return version === "v2.0.0"
+      ? concat([SignatureType.EOA, signature])
+      : signature;
+  };
+
+  const entryPointVersion: EntryPointVersion = (
+    AccountVersionRegistry[type][version] as any
+  ).entryPointVersion;
+
+  const entryPoint = {
+    abi: entryPointVersion === "0.6.0" ? entryPoint06Abi : entryPoint07Abi,
+    address:
+      entryPointVersion === "0.6.0"
+        ? ("0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789" as Address)
+        : ("0x0000000071727De22E5E9d8BAf0edAc6f37da032" as Address),
+    version:
+      entryPointVersion === "0.6.0" ? ("0.6" as const) : ("0.7" as const),
+  };
+
+  return await toSmartAccount({
+    getFactoryArgs,
+    client: client,
+    entryPoint: entryPoint as BaseLightAccountImplementation<
+      TSigner,
+      TLightAccountType,
+      TLightAccountVersion
+    >["entryPoint"],
+
+    async getAddress() {
+      return accountAddress;
+    },
+
+    async encodeCalls(calls) {
+      if (calls.length === 1) {
+        const call = calls[0];
+        return encodeFunctionData({
+          abi,
+          functionName: "execute",
+          args: [call.to, call.value ?? 0n, call.data ?? "0x"],
+        });
+      }
+
+      const [targets, values, datas] = calls.reduce(
+        (accum, curr) => {
+          accum[0].push(curr.to);
+          accum[1].push(curr.value ?? 0n);
+          accum[2].push(curr.data ?? "0x");
+          return accum;
+        },
+        [[], [], []] as [Address[], bigint[], Hex[]],
+      );
+
+      return encodeFunctionData({
+        abi,
+        functionName: "executeBatch",
+        args: [targets, values, datas],
+      });
+    },
+
+    async getStubSignature() {
+      const signature =
+        "0xfffffffffffffffffffffffffffffff0000000000000000000000000000000007aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa1c";
+
+      switch (version as string) {
```

Comment:
✨ could use a better type here and `assertNever`, but not too concerned since this is an old account type that prob won't have any new changes

Reply:
version should just work with this `as`. will fix